src/main/java/com/redhat/devtools/lsp4ij/LSPIJEditorUtils.java

```diff
@@ -0,0 +1,419 @@
+     * The latter is important for many editor support behaviors because you don't want to suspend something like end
+     * user typing to wait for a server to start.
+     *
+     * @param file the file
+     * @return true if the file is TextMate and is supported by a configured language server;
+     * otherwise false
+     */
+    @ApiStatus.Internal
+    public static boolean isSupportedTextMateFile(@NotNull PsiFile file) {
+        return isTextMateFile(file) && LanguageServersRegistry.getInstance().isFileSupported(file);
+    }
+
+    /**
+     * Determines whether or not the provided file is an abstract file type or TextMate file and is supported by a
+     * configured language server. The latter is important for many editor support behaviors because you don't want to
+     * suspend something like end user typing to wait for a server to start.
+     *
+     * @param file the file
+     * @return true if the file is either abstract file type or TextMate and is supported by a configured language
+     * server; otherwise false
+     */
+    @ApiStatus.Internal
+    public static boolean isSupportedAbstractFileTypeOrTextMateFile(@NotNull PsiFile file) {
+        return isSupportedAbstractFileTypeFile(file) || isSupportedTextMateFile(file);
+    }
+
+    private static boolean isAbstractFileTypeFile(@NotNull PsiFile file) {
+        return file.getFileType() instanceof AbstractFileType;
+    }
+
+    private static boolean isTextMateFile(@NotNull PsiFile file) {
+        return file.getFileType() instanceof TextMateFileType;
+    }
+
+    // Quote characters
+
+    private static final Set<Character> DEFAULT_QUOTE_CHARACTERS = Set.of('\'', '"', '`');
+
+    /**
+     * Returns the quote characters for the file. If it's a TextMate file or associated with an abstract file type,
+     * they're determined using the associated configuration data. If not, the defaults of single-quote, double-quote,
+     * and back-tick are returned.
+     *
+     * @param file the file for which quote characters should be returned
+     * @return the file's quote characters
+     */
+    @NotNull
+    @ApiStatus.Internal
+    public static Set<Character> getQuoteCharacters(@NotNull PsiFile file) {
+        Set<Character> quoteCharacters = isAbstractFileTypeFile(file) ? getAbstractFileTypeQuoteCharacters(file) :
+                isTextMateFile(file) ? getTextMateQuoteCharacters(file) :
+                        null;
+        if (quoteCharacters == null) {
+            quoteCharacters = DEFAULT_QUOTE_CHARACTERS;
+        }
+        return quoteCharacters;
+    }
+
+    @Nullable
+    private static Set<Character> getAbstractFileTypeQuoteCharacters(@NotNull PsiFile file) {
+        return isAbstractFileTypeFile(file) ? DEFAULT_QUOTE_CHARACTERS : null;
+    }
+
+    @Nullable
+    private static Set<Character> getTextMateQuoteCharacters(@NotNull PsiFile file) {
+        /* TODO: See TextMate versioning comment above
```

Comment:
This being commented out just results in the default quote characters being used.